packages/wrangler/src/module-collection.ts

```diff
@@ -51,12 +53,26 @@ export default function makeModuleCollector(): {
 
             return {
               path: "./" + fileName, // change the reference to the changed module
-              external: true, // mark it as external in the bundle
+              external: props.format === "modules", // mark it as external in the bundle
               namespace: "wrangler-module-collector-ns", // just a tag, this isn't strictly necessary
               watchFiles: [filePath], // we also add the file to esbuild's watch list
             };
           }
         );
+
+        if (props.format !== "modules") {
+          build.onLoad(
+            { filter: /.*\.(wasm)$/ },
+            async (args: esbuild.OnLoadArgs) => {
+              return {
+                contents: `export default ${args.path.replace(
+                  /[^a-zA-Z0-9_$]/g,
+                  "_"
+                )};`,
```

Comment:
A comment explaining the point of this code would be helpful here.

Reply:
Added some comments. Do they look ok to you?